fdbserver/DDTeamCollection.actor.cpp

```diff
@@ -2213,6 +2215,137 @@ class DDTeamCollectionImpl {
 		}
 	}
 
+	ACTOR static Future<Void> storageParamsUpdater(DDTeamCollection* self,
+	                                               StorageEngineParamSet newParams,
+	                                               Future<Void> wigglerInitialized) {
+		// always set the params from database configuration when DD starts
+		// there is no effect if it's consistent with storage servers
+		TraceEvent("DDTCStorageEngineParamsUpdaterStart")
+		    .detail("StoreType", self->configuration.storageServerStoreType.toString())
+		    .detail("NewParams", describe(newParams.getParams()));
+		// send update requests to all storages interfaces
+		std::vector<Future<ErrorOr<SetStorageEngineParamsReply>>> fs;
+		state std::map<UID, Future<ErrorOr<SetStorageEngineParamsReply>>> fmap;
+		for (const auto& [serverId, ss] : self->server_info) {
+			StorageServerInterface si = ss->getLastKnownInterface();
+			Future<ErrorOr<SetStorageEngineParamsReply>> f =
+			    si.setStorageEngineParams.tryGetReply(SetStorageEngineParamsRequest(newParams));
+			fs.push_back(f);
+			fmap[serverId] = f;
+		}
+		wait(waitForAll(fs) && wigglerInitialized);
+		state std::vector<Future<Void>> collection;
+		for (const auto& [serverId, f] : fmap) {
+			if (!f.isReady() || f.get().isError()) {
+				TraceEvent(SevWarnAlways, "DDTCSetStorageEngineParamsFailed")
+				    .detail("ServerId", serverId)
+				    .detail("Error", f.get().getError().code());
+				continue;
+			}
+			if (!self->server_info.contains(serverId)) {
+				TraceEvent(SevWarnAlways, "DDTCSetStorageEngineParamsServerRemoved").detail("ServerId", serverId);
+				continue;
+			}
+			SetStorageEngineParamsReply reply = f.get().get();
+			for (const auto& k : reply.result.applied) {
+				newParams.getParams().contains(k)
+				    ? self->configuration.storageEngineParams.set(k, newParams.getParams().at(k))
+				    : self->configuration.storageEngineParams.clear(k);
+				TraceEvent(SevInfo, "StorageEngineParamApplied")
+				    .detail("ServerId", serverId)
+				    .detail("Param", k)
+				    .detail("Value", newParams.getParams().contains(k) ? newParams.getParams().at(k) : "");
+			}
```

Comment:
Because there are many similar trace code below, I think it can make the code concise to add an inline function doing this trace pattern.

Reply:
sure, I will update it.

Reply:
Done.